org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NameEnvironmentAnswerListenerTest.java

```diff
@@ -0,0 +1,178 @@
+				URI uri = answer.getBinaryType().getURI();
+				this.answeredFileNames.add(uri.toString());
+				return;
+			} else if(answer.getCompilationUnit() != null) {
+				fileName = answer.getCompilationUnit().getFileName();
+			} else if(answer.getSourceTypes() != null && answer.getSourceTypes().length > 0) {
+				fileName = answer.getSourceTypes()[0].getFileName(); // the first type is guaranteed to be the requested type
+			} else if(answer.getResolvedBinding() != null) {
+				fileName = answer.getResolvedBinding().getFileName();
+			}
+			if (fileName != null) this.answeredFileNames.add(new String(fileName));
+		}
+	}
+
+	public void testNameEnvironmentAnswerListener() throws IOException {
+		String path = LIB_DIR;
+		if(!path.endsWith(File.separator)) {
+			path += File.separator;
+		}
+		String libPath = path + "lib.jar";
+		Util.createJar(
+				new String[] {
+						"p/Color.java",
+						"package p;\n" +
+						"public enum Color {\n" +
+						"	R, Y;\n" +
+						"	public static Color getColor() {\n" +
+						"		return R;\n" +
+						"	}\n" +
+						"}",
+					},
+				libPath, JavaCore.VERSION_17);
+
+		String unusedLibPath = path + "lib_unused.jar";
+		Util.createJar(
+				new String[] {
+						"p2/Color.java",
+						"package p2;\n" +
+						"public enum Color {\n" +
+						"	R, Y;\n" +
+						"	public static Color getColor() {\n" +
+						"		return R;\n" +
+						"	}\n" +
+						"}",
+					},
+				unusedLibPath, JavaCore.VERSION_17);
+
+		String srcDir =  path + "src";
+		String[] pathsAndContents =
+				new String[] {
+					"s/X.java",
+					"package s;\n" +
+					"import p.Color;\n" +
+					"public class X {\n" +
+					"	public static final Color MY = Color.R;\n" +
+					"}"
+				};
+		Util.createSourceDir(pathsAndContents, srcDir);
+
+		List<String> classpath = new ArrayList<>(Arrays.asList(getDefaultClassPaths()));
+		classpath.add(libPath);
+		classpath.add(unusedLibPath);
+
+		File outputDirectory = new File(Util.getOutputDirectory());
+		if (!outputDirectory.isDirectory()) {
+			outputDirectory.mkdirs();
+		}
+
+		List<String> ecjArguments = new ArrayList<>();
+
+		ecjArguments.add("-classpath");
+		ecjArguments.add(classpath.stream()
+				.map(jar -> jar.equals(unusedLibPath) ? String.format("%s[-**/*]", jar) : jar)
+				.collect(joining(File.pathSeparator)));
+
+
+		ecjArguments.add("-d");
+		ecjArguments.add(outputDirectory.getAbsolutePath());
+
+		ecjArguments.add("--release");
+		ecjArguments.add("17");
+
+		ecjArguments.add(srcDir+ File.separator + "s"+ File.separator + "X.java");
+
+		EclipseBatchCompiler compiler;
+		boolean compileOK;
+		File logFile = new File(outputDirectory, "compile.log");
+		try(PrintWriter log = new PrintWriter(new FileOutputStream(logFile))) {
+			compiler = new EclipseBatchCompiler(log);
+			compileOK = compiler.compile(ecjArguments.toArray(new String[ecjArguments.size()]));
+		}
+		String logOutputString = Util.fileContent(logFile.getAbsolutePath());
+
+		if(!compileOK) {
```

Comment:
Could we please use my version?
```
		EclipseBatchCompiler compiler;
		File logFile = new File(outputDirectory, "compile.log");
		try(PrintWriter log = new PrintWriter(new FileOutputStream(logFile))) {
			compiler = new EclipseBatchCompiler(log);
			boolean compileOK;
			compileOK = compiler.compile(ecjArguments.toArray(new String[ecjArguments.size()]));
			if(!compileOK) {
				String logOutputString = Util.fileContent(logFile.getAbsolutePath());
				Assert.fail("Compile failed, output: '" + logOutputString + "'");
			}
		}
```
With the current one, if you break compilation, test reports only `java.lang.AssertionError: Compile failed but did not throw exception!` but in my version we would see immediately **why** :
```
java.lang.AssertionError: Compile failed, output: '----------
1. ERROR in /tmp/comptest/run.1718287287470/lib/src/s/X.java (at line 1)
	pack age s;
	^^^^^^^^
Syntax error on tokens, they can be merged to form package
----------
1 problem (1 error)```